frontend/catalyst/api_extensions/differentiation.py

```diff
@@ -801,8 +801,6 @@ def _make_jaxpr_check_differentiable(
     with mark_gradient_tracing(method):
         jaxpr, shape = jax.make_jaxpr(f, return_shape=True)(*args, **kwargs)
     _, out_tree = tree_flatten(shape)
-    assert len(jaxpr.eqns) == 1, "Expected jaxpr consisting of a single function call."
-    assert jaxpr.eqns[0].primitive == func_p, "Expected jaxpr consisting of a single function call."
```

Comment:
Probably similar to my other comment, is this addressing a second bug regarding derivatives and dynamic one shot, or is it related to ZNE somehow? :)

Reply:
no this is not related to ZNE, it happens when we use derivatives with dynamic one-shot